sympy/core/function.py

```diff
@@ -428,10 +428,11 @@ def __new__(cls, *args, **options):
         if not evaluate or not isinstance(result, cls):
             return result
 
-        pr = max(cls._should_evalf(a) for a in result.args)
-        pr2 = min(cls._should_evalf(a) for a in result.args)
-        if pr2 > 0:
-            return result.evalf(mlib.libmpf.prec_to_dps(pr))
+        if len(result.args) > 0:
```

Comment:
That is true. Not quite sure if it would be necessary better, I tend to try to avoid using autocast to boolean. But then again, I don't see how it could hurt here.
So if someone else seconds this motion I will change it.